cmd/internal/flowmode/cmd_run.go

```diff
@@ -334,6 +339,48 @@ func (fr *flowRun) Run(configFile string) error {
 		return err
 	}
 
+	// Reload on file change
+	if fr.configAutoReload {
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			level.Error(l).Log("msg", "failed to reload config", "err", err)
+		}
+		defer watcher.Close()
+
+		// Stop reloading on SIGINT and SIGTERM
+		sigCh := make(chan os.Signal, 1)
+		signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+		defer signal.Stop(sigCh)
+
+		go func() {
```

Comment:
Can you synchronize this goroutine with the waitgroup so we ensure it closes before Flow exits?

```suggestion
		wg.Add(1)
		go func() {
			defer wg.Done()
```